deepchem/models/dft/nnxc.py

```diff
@@ -0,0 +1,183 @@
+from abc import abstractmethod
+from typing import Union
+import torch
+from dqc.utils.datastruct import ValGrad, SpinParam
+from dqc.api.getxc import get_xc
+
+
+class BaseNNXC(torch.nn.Module):
+    """
+    Base class for the NNLDA and HybridXC classes.
```

Comment:
Can you add more detail on when we would want to use this base class and what capabilities it provides? A good paragraph of explanation would be my rule of thumb